nebula_ros/src/hesai/hesai_ros_wrapper.cpp

```diff
@@ -292,6 +380,22 @@ rcl_interfaces::msg::SetParametersResult HesaiRosWrapper::OnParameterChange(
     return result;
   }
 
+  // Set calibration (if any) only once all parameters have been validated
+  if (new_calibration_ptr) {
+    decoder_wrapper_->OnCalibrationChange(new_calibration_ptr);
+    RCLCPP_INFO_STREAM(get_logger(), "Changed calibration to '" << new_cfg.calibration_path << "'");
+  }
+
+  if (
+    new_calibration_ptr && hw_interface_wrapper_ &&
+    sensor_cfg_ptr_->sensor_model != drivers::SensorModel::HESAI_PANDARAT128) {
+    auto status = hw_interface_wrapper_->HwInterface()->checkAndSetLidarRange(*new_calibration_ptr);
+    if (status != Status::OK) {
+      RCLCPP_ERROR_STREAM(
+        get_logger(), "Could not change FoV but config already changed: " << status);
```

Comment:
This error message is a little contradictory. Is it because everything has been set except range which failed?
In which case, something like "Sensor configuration updated, but FoV could not be set" might be better.

Reply:
Re-worded in 7d4effa